packages/language-server/src/worker.ts

```diff
@@ -0,0 +1,9 @@
+import { doSemanticAnalysis } from '@neo4j-cypher/language-support';
+import { parentPort } from 'worker_threads';
+
+parentPort.on('message', (value: { query: string; port: MessagePort }) => {
```

Comment:
Dummy question. Why do we have a parentPort here and a `self.onmessage` in the react-codemirror package?

Is it due to node vs browser approach?

Reply:
yes